### PR TITLE
Rework the behavior of favicon search

### LINKED ISSUE #5838: [BUG] Sites with non-absolute paths to favicons don't load
**Describe the bug**

Sites that have favicon link href's using a relative or absolute path without a domain don't load.

Examples:
  * https://www.androidpolice.com/ - uses absolute path with no host: `/public/build/images/favicon-48x48.5475dbfb.png`
  * https://www.synology.com/en-us/company/news - uses base element with relative path: `img/icon/touch-icon-57.png`

**To Reproduce**
1. Subscribe to one of the above sites
2. Instead of showing the site's favicon on feed items, it shows the generic one.

**Expected behavior**
The favicon referenced on the site should be shown.

**Environment information (please complete the following information):**
- Device: Any
- OS: Any
- Browser: Any
- FreshRSS version: 1.22.1
- Database version: SQLite
- PHP version: PHP 8.2
- Installation type: Manual

**Additional context**
I've been testing a fix that has worked on all of the sites I've tested it with. I'll submit a PR.

### PR DESCRIPTION
**Pull request by jaden**
Closes #5838 

Changes proposed in this pull request:

- Use xpath queries to reduce the filtering necessary.
- Don't call checkUrl since the href value will not always be a full URL.
- Check for a base path in the HTML to be used if the favicon href doesn't start with a leading slash.

How to test the feature manually:

1. Subscribe to sites with favicon link hrefs that don't include the domain.
2. Verify that the correct favicon is loaded.

Pull request checklist:

- [X] clear commit messages
- [X] code manually tested
- [ ] unit tests written (optional if too hard)
- [ ] documentation updated

[Additional information can be found in the documentation](https://github.com/FreshRSS/FreshRSS/tree/edge/docs/en/developers/04_Pull_requests.md).

### CODE REVIEW

**Comment by Alkarex** [comment]
Thanks! Looks good.
Try to run `make test-all`

File: lib/favicons.php [hunk 1]
```diff
@@ -68,34 +68,33 @@ function downloadHttp(string &$url, array $curlOptions = []): string {
...
+	$baseElements = $xpath->query('//base[@href]');
+	$baseUrl = ($baseElements->length > 0) ? $baseElements->item(0)->getAttribute('href') : $url;
+
+	foreach ($links as $link) {
+		$href = trim($link->getAttribute('href'));
+		$urlParts = parse_url($url);
+
+		// Handle protocol-relative URLs by adding the current URL's scheme
+		if (substr($href, 0, 2) === '//') {
+			$href = $urlParts['scheme'] . '://' . $href;
```
Comment (Alkarex) [inline lib/favicons.php]: Example of fix (several of the other errors are of the same type):

```suggestion
			$href = ($urlParts['scheme'] ?? 'https') . '://' . $href;
```

File: lib/favicons.php [hunk 2]
```diff
@@ -68,34 +68,33 @@ function downloadHttp(string &$url, array $curlOptions = []): string {
...
+		return '';
+	}
+
+	$xpath = new DOMXPath($dom);
+	$links = $xpath->query('//link[@href][translate(@rel, "ABCDEFGHIJKLMNOPQRSTUVWXYZ", "abcdefghijklmnopqrstuvwxyz")="shortcut icon"'
+		. ' or translate(@rel, "ABCDEFGHIJKLMNOPQRSTUVWXYZ", "abcdefghijklmnopqrstuvwxyz")="icon"]');
+
+	// Use the base element for relative paths, if there is one
+	$baseElements = $xpath->query('//base[@href]');
+	$baseUrl = ($baseElements->length > 0) ? $baseElements->item(0)->getAttribute('href') : $url;
```
Comment (Alkarex) [inline lib/favicons.php]: Here you need to test that `$xpath->query` did not return `false`, and also test that your first element is a `DOMElement` (and not only a `DOMNode`)

Comment (jaden) [reply lib/favicons.php]: Thanks for the tips, much appreciated.

**Comment by Alkarex** [comment]
Good improvement 👍🏻 
Please add a line for you https://github.com/FreshRSS/FreshRSS/blob/edge/CREDITS.md

**Pull request merged**